packages/plugin-legacy/index.js

```diff
@@ -97,7 +97,7 @@ function viteLegacyPlugin(options = {}) {
     apply: 'build',
```

Comment:
I guess ideally we could have some feature extends apply, like

```ts
apply: 'client-build' | 'ssr-build'
```

but in that case, maybe a function would be more flexible:

```ts
apply(config) {
  return config.command === 'build' && !config.build.ssr
}
```

🤔

Reply:
I'm voting for a function; I can see other scenarios where an apply function would be helpful.